build-tools/tasks/generate-i18n-messages.js

```diff
@@ -0,0 +1,51 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+const fs = require('fs');
+const path = require('path');
+const { parse } = require('@formatjs/icu-messageformat-parser');
+
+const { targetPath } = require('../utils/workspace');
+const { writeFile } = require('../utils/files');
+
+const namespace = '@cloudscape-design/components';
+
+const destinationDir = path.join(targetPath, 'components/internal/i18n/messages');
+const declarationFile = `import { I18nProviderProps } from "../provider";
+const messages: I18nProviderProps.Messages;
+export default messages;
+`;
+
+module.exports = function generateI18nMessages() {
+  const messagesDir = path.resolve(__dirname, '../../src/internal/i18n/messages');
+  const files = fs.readdirSync(messagesDir);
+
+  const allParsedMessages = {};
+
+  for (const fileName of files) {
+    const filePath = path.join(messagesDir, fileName);
+    const messages = require(filePath);
+    const [subset, locale] = fileName.split('.');
+
+    const parsedMessages = Object.fromEntries(
+      Object.entries(messages).map(([component, keys]) => [
+        component,
+        Object.fromEntries(Object.entries(keys).map(([key, icuMessage]) => [key, parse(icuMessage, { locale })])),
+      ])
+    );
+    allParsedMessages[locale] = { ...(allParsedMessages[locale] ?? {}), ...parsedMessages };
```

Comment:
why do we need this part `allParsedMessages[locale]`? There could be multiple files for the same locale? Like for individual components in the future?